susemanager-utils/testing/docker/scripts/schema_idempotency_test_pgsql.py

```diff
@@ -0,0 +1,225 @@
+                        default="/etc/sysconfig/rhn/schema-upgrade")
+    parser.add_argument("-p", "--from-pr", action="store", dest="prFile",
+                        help="Check files with changes from a PR by rerunning added or changed files. The value is the path to a JSON file with PR information generated by gitarro")
+    parser.add_argument("-v", "--from-version", action="store", dest="version",
+                        help="Check all files since a base version (for example 4.0.0)")
+    parser.add_argument("-u", "--upgrade-script", action="store", dest="upgradeScript",
+                        help="Path where the spacewalk-schema-upgrade script is (default: /manager/schema/spacewalk/spacewalk-schema-upgrade)",
+                        default="/manager/schema/spacewalk/spacewalk-schema-upgrade")
+    args = parser.parse_args()
+    if args.schemaPath is None:
+        raise RuntimeError(
+            "You need to specify the path where the schema upgrades are")
+    if not directory_exists(args.schemaPath):
+        raise RuntimeError("Directory %s does not exist" % args.schemaPath)
+    if args.prFile is None and args.version is None:
+        raise RuntimeError(
+            "One of --from-pr or --from-version must be specified")
+    if args.prFile and args.version:
+        raise RuntimeError(
+            "Only one of --from-pr or --from-version must be specified")
+    if args.prFile and not file_exists(args.prFile):
+        raise RuntimeError("File %s does not exist" % prFile)
+    if args.upgradeScript is None:
+        raise RuntimeError(
+            "You need to specify the path where the spacewalk-schema-upgrade script is")
+    if not file_exists(args.upgradeScript):
+        raise RuntimeError(
+            "The file %s for the upgrade script does not exist" % args.upgradeScript)
+    print("============================================================")
+    print("                     Idempotency test                       ")
+    print("============================================================")
+    # Exclude tables and sequences that are not updated by the SQL scripts (version control)
+    excludedTables = ["pg_catalog.setval", "public.rhnpackageevr",
+                      "public.rhn_pkg_evr_seq", "public.rhnversioninfo"]
+    # Calculate new version for fake migrations
+    newVersion = find_next_version(args.schemaPath)
+    create_fake_migration_path(
+        args.schemaPath, newVersion, args.prFile, args.version)
+    manage_postgresql('start')
+    initial_dump = "/tmp/initial_dump.sql"
+    dump_database(initial_dump, excludedTables)
+    run_upgrade(args.upgradeScript, args.schemaPath, newVersion)
+    migrated_dump = "/tmp/migrated_dump.sql"
+    dump_database(migrated_dump, excludedTables)
+    manage_postgresql('stop')
+    diff = False
+    for line in diff_dumps(initial_dump, migrated_dump):
+        diff = True
+        print(line)
+    if diff:
```

Comment:
i would vote for GuardClause `if ! diff contiune`

Reply:
Done and testing.